### DISCUSSION TITLE
Dart & Flutter - Unable To Validate Email Address: Invalid Format

### DISCUSSION THREAD

**Discussion by rohan-bansal** [supabase/supabase#2012] [Questions]
Hi!

I'm writing a Flutter app interfacing with Supabase to authenticate a user.

I'm validating the form fields in a function like so:
```dart
    var email = _emailController.text;
    if (!email.contains('@')) {
      _showDialog(context, title: 'Error', message: 'Invalid email!');
      return false;
    }

    var pwd = _passwordController.text;
    if (pwd.isEmpty || pwd.length < 6) {
      _showDialog(context,
          title: 'Error',
          message: 'Password is empty or is not at least 6 characters long!');
      return false;
    }

    return true;
```

And I'm passing it to Supabase to authenticate like this:

```dart
    final result = await GetIt.instance<SupabaseClient>()
        .auth
        .signUp(_emailController.text, _passwordController.text);


    if (result.data != null) {
      Navigator.pushReplacementNamed(context, '/login');
      _showDialog(context, title: 'Success', message: 'Register Successful');
    } else {
      _showDialog(context, title: 'Error', message: result.error.message);
    }
```

However, the sign up fails and `result.error.message` is "Unable to validate email address: invalid format". I've made sure my email is correct, and even printed the value out before passing it to the auth function (of the form `test1234test@gmail.com` and `test@email.com`). 

Any ideas (ignoring any bad code practice)? Help is appreciated!

**Comment by inian** [comment]
Our Gotrue server validates the email and the error seems to be coming from there. Usually I run into this error if there is a trailing space in the email being validated. 

You can see your Gotrue logs from here - `/project/your-project-ref/auth/settings`. Can you see some error there? 

Gotrue uses [this third party](https://github.com/badoux/checkmail/blob/a155f10fefd532ab05f965c7486e50e20a832bd0/checkmail.go#L37) library to validate emails
